org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/JSEditorAssociationOverride.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Victor Rubezhny (Red Hat Inc.) - initial implementation
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.ui.IEditorDescriptor;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorRegistry;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.genericeditor.ExtensionBasedTextEditor;
+import org.eclipse.ui.internal.genericeditor.GenericEditorWithContentTypeIcon;
+import org.eclipse.ui.internal.genericeditor.GenericEditorWithIconAssociationOverride;
+
+public class JSEditorAssociationOverride extends GenericEditorWithIconAssociationOverride {
+	private Map<String, IEditorDescriptor> defaultDescriptorMap = new HashMap<>();
+	
+	@Override
+	public IEditorDescriptor overrideDefaultEditor(IEditorInput editorInput, IContentType contentType,
+			IEditorDescriptor editorDescriptor) {
+		return editorInput != null ? overrideDefaultEditor(editorInput.getName(), contentType, editorDescriptor)
+				: editorDescriptor;
+	}
+
+	@Override
+	public IEditorDescriptor overrideDefaultEditor(String fileName, IContentType contentType,
+			IEditorDescriptor editorDescriptor) {
+		return getDefaultEditorDescriptorForFile(editorDescriptor, fileName);
+	}
+
+	// Workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=559907
+	private IEditorDescriptor getDefaultEditorDescriptorForFile(IEditorDescriptor defaultDescriptor, String fileName) {
+		if (!ExtensionBasedTextEditor.GENERIC_EDITOR_ID.equals(defaultDescriptor.getId())
+				&& fileName != null && !fileName.isEmpty()) {
+			int index = fileName.lastIndexOf('.');
+			if (index > -1) {
+				String extension = fileName.substring(index);
+				if (".js".equalsIgnoreCase(extension.toLowerCase())) {
+					IEditorRegistry editorReg = PlatformUI.getWorkbench()
+							.getEditorRegistry();
+					
+					IEditorDescriptor genericEditorDescriptor = editorReg.findEditor(ExtensionBasedTextEditor.GENERIC_EDITOR_ID);
```

Comment:
This could be made a final field, defined int he constructor, as we're actually reusing it all the time.